doc/whats_new.rst

```diff
@@ -251,6 +252,12 @@ API changes summary
       :func:`sklearn.model_selection.cross_val_predict`.
       :issue:`2879` by :user:`Stephen Hoover <stephen-hoover>`.
 
+    - In :class:`pipeline.Pipeline`, ``steps`` has been changed by ``steps`` and
```

Comment:
I think this is better as:
```
In the future, the estimators in the ``steps`` and ``named_steps``
attributes will no longer have their ``fit()`` methods called directly.
Users will have to access fitted Pipeline steps in ``steps_``
and ``named_steps_``.
```

I think we should make this an extra long deprecation and note the deprecation period in what's new.